my-code/frontend/src/hooks/useLike.ts

```diff
@@ -0,0 +1,48 @@
+import React from "react";
+
+export default function useLike(imdbID: string) {
+  const [liked, setLiked] = React.useState<boolean>(() => {
+    const likedMovies = localStorage.getItem("@App/likedMovies");
+    let parsed;
+    try {
+      parsed = JSON.parse(likedMovies || "");
+    } catch {
+      parsed = {};
+    }
+    // Set liked initial state
+    if (parsed[imdbID]) {
+      return true;
+    }
+    return false;
+  });
+
+  React.useEffect(() => {
+    const likedMovies = localStorage.getItem("@App/likedMovies");
+    let parsed;
+
+    // Avoid JSON.Parse error if the stringified object is malformed for some reason
+    try {
+      parsed = JSON.parse(likedMovies || "");
+    } catch {
+      console.log("Failed to parse JSON...");
+      parsed = {};
+    }
+
+    const newLikedMovies = {
+      ...parsed,
+      [imdbID]: liked,
+    };
+
+    // Save likes to local storage
+    localStorage.setItem("@App/likedMovies", JSON.stringify(newLikedMovies));
+  }, [liked]);
+
+  const handleLike = (e: React.MouseEvent<HTMLButtonElement>) => {
```

Comment:
IMO, `toggleLike` would have been more descriptive.
also, it's a bit strange to capture the mouse event and prevent default here when it seems totally unrelated to the "liking" action.
what if someone wants to use some other trigger to toggle this behaviour?

`preventDefault` should the the button's responsibility, if needed (maybe you could even try to do `<button type="button"` which should prevent the route change)